protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -75,8 +108,7 @@
     private boolean isDeletedByOtherSession = false;
     private boolean sizeChanged = false;
     private boolean silentFlagChanges = false;
-    private final Flags applicableFlags;
-    private boolean applicableFlagsChanged;
+    private ApplicableFlags applicableFlags;
```

Comment:
Don't we need to be volatile? Or defend it with an atomic ref?

Reply:
It doesn't deal with thread-safety yet, it refactor datastructure for being able to reach thread safety in a later step. So volatile is of no use (and BTW it would not be exercised because we don't have the tests yet)